sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1804,15 +1804,15 @@ void ProgramManager::cacheKernelImplicitLocalArg(
 DeviceKernelInfo &ProgramManager::getOrCreateDeviceKernelInfo(
     const CompileTimeKernelInfoTy &Info) {
   std::lock_guard<std::mutex> Guard(m_DeviceKernelInfoMapMutex);
-  auto [Iter, Inserted] =
-      m_DeviceKernelInfoMap.try_emplace(KernelNameStrT{Info.Name.data()}, Info);
+  auto [Iter, Inserted] = m_DeviceKernelInfoMap.try_emplace(
+      std::string_view{Info.Name.data()}, Info);
```

Comment:
I had tried to add `detail::string_view::operator std::string_view` but it resulted in ambiguities with some comparison operators (I think). Now that we can remove old ones, maybe a better fix would be to change `detail::string_view` to have this conversion and rely on it happening implicitly for comparisons.

Reply:
Right, I recall running into something along those lines. On this line in particular, though, there's no conversion to be made at all and I just didn't catch that at first.